src/pocketmine/entity/projectile/Arrow.php

```diff
@@ -74,6 +77,10 @@ public function onUpdate($currentTick){
 
 		if($this->onGround or $this->hadCollision){
 			$this->setCritical(false);
+			if($this->sound === true and $this->level !== null){ //Prevents error of $this->level returning null
```

Comment:
If the arrow is actually closed before it reaches this point for whatever reason, the correct solution would be to check if the entity is closed, not whether its level is null.